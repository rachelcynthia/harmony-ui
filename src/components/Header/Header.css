nav {
  position: sticky;
  display: flex;
  top: 0;
  width: 100%;
  overflow: auto;
  height: auto;
  background: #000;
  padding: 8px 0px;
}
nav .left-links {
  flex: 2 1 200px;
  padding: 0px 10px;
  margin-top: -4px;
  margin-bottom: -5px;
}
nav .left-links img {
  filter: brightness(160%);
  cursor: pointer;
}
nav .centered-links {
  flex-basis: 50%;
}
nav .right-links {
  padding-right: 20px;
}
nav .links {
  display: inline-block;
  text-align: center;
  padding: 2px;
  font-size: 30px;
  text-decoration: none;
  font-weight: bolder;
  font-family: Georgia, 'Times New Roman', Times, serif;
  color: #cb6ce6;
}
.circle {
  background: #cb6ce6;
  border: none;
  cursor: pointer;
  border-radius: 50%;
  width: 40px;
  height: 40px;
}
