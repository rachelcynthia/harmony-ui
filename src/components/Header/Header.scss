nav {
    position: sticky;
    display: flex;
    top:0;
    width: 100%;
    overflow: auto;
    height: auto;
    background: #000;
    padding: 8px 0px;
    .left-links{
      flex:2 1 200px;
      padding: 0px 10px;
      margin-top: -4px;
      margin-bottom: -5px;
      img {
        filter: brightness(160%);
        cursor: pointer;
      }
   }
   .centered-links{
    flex-basis: 50%;
   }
   .right-links{
     padding-right: 10px;
   }
   .links {
    display: inline-block;
    text-align: center;
    padding: 2px;
    font-size: 30px;
    text-decoration: none;
    font-weight: bolder;
    font-family: Georgia, 'Times New Roman', Times, serif;
    color: #CB6CE6;
 }
}
.circle {
  background: #CB6CE6;
  border: none;
  cursor: pointer;
  border-radius: 50%;
  width: 40px;
  height: 40px;
}
  