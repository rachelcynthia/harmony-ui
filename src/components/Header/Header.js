import React from 'react';
import './Header.css'
import profile from '../../assets/person-fill.svg'
import logo from '../../assets/logo.png'

const Header = ({ isLoggedIn }) => {
    // render() {
    return (
        <nav>
            <div className="left-links">
                <img src={logo} alt="Logo" width="50" height="50" />
            </div>
            <div className="centered-links">
                <a className="links">Harmony</a>
            </div>
            {isLoggedIn === true ?
                <div className="right-links" data-testid="profile">
                    <button className="circle"><img src={profile} alt="Profile" width="25" height="25" /></button>
                </div> : <div className="right-links" style={{ width: "50px" }}></div>}
        </nav>
    );
    // }
}

export default Header