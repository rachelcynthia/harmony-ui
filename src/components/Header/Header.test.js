import { render } from '@testing-library/react';
import React from 'react';
import TestRenderer from 'react-test-renderer';
import Header from './Header';

it('renders correctly', () => {
    const header = TestRenderer.create(<Header />).toJSON();
    expect(header).toMatchSnapshot();
});

it('should not render profile pic icon when isLoggedIn is false', () => {
    const header = render(<Header isLoggedIn={false} />);
    expect(() => header.getByTestId("profile")).toThrow()
})