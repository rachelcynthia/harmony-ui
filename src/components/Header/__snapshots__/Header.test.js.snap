// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav>
  <div
    className="left-links"
  >
    <img
      alt="Logo"
      height="50"
      src="test-file-stub"
      width="50"
    />
  </div>
  <div
    className="centered-links"
  >
    <a
      className="links"
    >
      Harmony
    </a>
  </div>
  <div
    className="right-links"
    style={
      Object {
        "width": "50px",
      }
    }
  />
</nav>
`;
